astropy/units/physical.py

```diff
@@ -311,6 +313,19 @@ def __init__(self, unit, physical_types):
     def __iter__(self):
         yield from self._physical_type_list
 
+    def __getattr__(self, attr):
+        if attr in dir(str):
+            warning_message = (
+                f"support for accessing str attributes such as {attr} "
+                "from PhysicalType instances will be removed in a "
+                "subsequent release."
+            )
```

Comment:
My preference is also for black but 
https://docs.astropy.org/en/stable/development/codeguide.html?highlight=black#coding-style-conventions